Error release 3.2.0 mingw

Error:

```
[24/25] Linking CXX shared library libprecice.dll
FAILED: libprecice.dll libprecice.dll.a
C:\Windows\system32\cmd.exe /C "cd . && C:\msys64\ucrt64\bin\g++.exe -march=nocona -msahf -mtune=generic -O2 -pipe -Wno-attributes -O3 -DNDEBUG  -Wl,--export-all-symbols -shared -o libprecice.dll -Wl,--out-implib,libprecice.dll.a -Wl,--major-image-version,3,--minor-image-version,2 CMakeFiles/preciceCore.dir/Unity/unity_partition_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_action_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_mesh_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_io_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_profiling_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_mapping_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_time_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_m2n_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_math_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_cplscheme_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_logging_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_precice_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_acceleration_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_com_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_query_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_utils_cxx.cxx.obj CMakeFiles/preciceCore.dir/Unity/unity_xml_cxx.cxx.obj CMakeFiles/preciceCore.dir/src/precice/impl/versions.cpp.obj  C:/msys64/ucrt64/lib/libboost_log_setup-mt.dll.a  C:/msys64/ucrt64/lib/libboost_log-mt.dll.a  C:/msys64/ucrt64/lib/libboost_filesystem-mt.dll.a  C:/msys64/ucrt64/lib/libboost_serialization-mt.dll.a  C:/msys64/ucrt64/lib/libboost_program_options-mt.dll.a  C:/msys64/ucrt64/lib/libboost_thread-mt.dll.a  C:/msys64/ucrt64/lib/libboost_date_time-mt.dll.a  C:/msys64/ucrt64/lib/libboost_atomic-mt.dll.a  -lsynchronization  C:/msys64/ucrt64/lib/libboost_chrono-mt.dll.a  C:/msys64/ucrt64/lib/libboost_container-mt.dll.a  C:/msys64/ucrt64/lib/libboost_unit_test_framework-mt.dll.a  C:/msys64/ucrt64/lib/libxml2.dll.a  thirdparty/fmt/libfmtlib-static.a  C:/msys64/ucrt64/lib/libmsmpi.dll.a  C:/msys64/ucrt64/lib/libpython3.12.dll.a  C:/msys64/ucrt64/lib/libpython3.12.dll.a  -lws2_32 && cd ."
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/preciceCore.dir/Unity/unity_partition_cxx.cxx.obj:unity_partition_cxx.cxx:(.text$_ZN7precice5ErrorD1Ev[_ZN7precice5ErrorD1Ev]+0x3): undefined reference to `__imp__ZTVN7precice5ErrorE'
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/preciceCore.dir/Unity/unity_partition_cxx.cxx.obj:unity_partition_cxx.cxx:(.text$_ZN7precice5ErrorD0Ev[_ZN7precice5ErrorD0Ev]+0x8): undefined reference to `__imp__ZTVN7precice5ErrorE'
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/preciceCore.dir/Unity/unity_partition_cxx.cxx.obj:unity_partition_cxx.cxx:(.text$_ZN7precice7logging16logErrorAndThrowINS_5ErrorEEEvRNS0_6LoggerENS0_11LogLocationERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7precice7logging16logErrorAndThrowINS_5ErrorEEEvRNS0_6LoggerENS0_11LogLocationERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x56): undefined reference to `__imp__ZTVN7precice5ErrorE'
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/preciceCore.dir/Unity/unity_math_cxx.cxx.obj:unity_math_cxx:(.text+0x2867): undefined reference to `__imp__ZTVN7precice5ErrorE'
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

Same:

```
-- Build files have been written to: C:/msys64/usr/local/pkg_precice/src/build-UCRT64-shared
[1/145] C:\Windows\system32\cmd.exe /C "cd /D C:\m...cice-3.2.0/cmake/cmake_refresh_git_revision.cmake
-- Found Git: C:/msys64/usr/bin/git.exe (found version "2.49.0")
fatal: not a git repository (or any of the parent directories): .git
-- Revision status: Detection failed
[144/145] Linking CXX shared library libprecice.dll
FAILED: libprecice.dll libprecice.dll.a
C:\Windows\system32\cmd.exe /C "cd . && C:\msys64\ucrt64\bin\g++.exe -march=nocona -msahf -mtune=generic -O2 -pipe -Wno-attributes -O3 -DNDEBUG  -Wl,--export-all-symbols -shared -o libprecice.dll -Wl,--out-implib,libprecice.dll.a -Wl,--major-image-version,3,--minor-image-version,2 @CMakeFiles\precice.rsp && cd ."
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/preciceCore.dir/src/acceleration/AitkenAcceleration.cpp.obj:AitkenAcceleration.cpp:(.text+0xcf7): undefined reference to `__imp__ZTVN7precice5ErrorE'
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/preciceCore.dir/src/acceleration/AitkenAcceleration.cpp.obj:AitkenAcceleration.cpp:(.text$_ZN7precice5ErrorD1Ev[_ZN7precice5ErrorD1Ev]+0x3): undefined reference to `__imp__ZTVN7precice5ErrorE'
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/preciceCore.dir/src/acceleration/AitkenAcceleration.cpp.obj:AitkenAcceleration.cpp:(.text$_ZN7precice5ErrorD0Ev[_ZN7precice5ErrorD0Ev]+0x8): undefined reference to `__imp__ZTVN7precice5ErrorE'
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/preciceCore.dir/src/acceleration/BaseQNAcceleration.cpp.obj:BaseQNAcceleration.cpp:(.text$_ZN7precice7logging16logErrorAndThrowINS_5ErrorEEEvRNS0_6LoggerENS0_11LogLocationERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7precice7logging16logErrorAndThrowINS_5ErrorEEEvRNS0_6LoggerENS0_11LogLocationERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x56): undefined reference to `__imp__ZTVN7precice5ErrorE'
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/preciceCore.dir/src/acceleration/ConstantRelaxationAcceleration.cpp.obj:ConstantRelaxationAcceleration.cpp:(.text+0x18e): undefined reference to `__imp__ZTVN7precice5ErrorE'
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/preciceCore.dir/src/com/config/CommunicationConfiguration.cpp.obj:CommunicationConfiguration.cpp:(.text+0x7ca): more undefined references to `__imp__ZTVN7precice5ErrorE' follow
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

with `-DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON`:

```
[144/145] Linking CXX shared library libprecice.dll
FAILED: libprecice.dll libprecice.dll.a
C:\WINDOWS\system32\cmd.exe /C "cd . && C:\msys64\ucrt64\bin\g++.exe -march=nocona -msahf -mtune=generic -O2 -pipe -Wno-attributes -O3 -DNDEBUG -flto=auto -fno-fat-lto-objects  -Wl,--export-all-symbols -shared -o libprecice.dll -Wl,--out-implib,libprecice.dll.a -Wl,--major-image-version,3,--minor-image-version,2 @CMakeFiles\precice.rsp && cd ."
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot export _ZN3fmt3v106detail12write_paddedILNS0_5align4typeE2ENS0_8appenderEcRZNS1_14do_write_floatIS5_NS1_14big_decimal_fpEcNS1_14digit_groupingIcEEEET_SA_RKT0_RKNS0_12format_specsIT1_EENS1_11float_specsENS1_10locale_refEEUlS5_E3_EESB_SB_SI_yyOT2_: symbol wrong type (4 vs 3)
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot export _ZN3fmt3v106detail12write_paddedILNS0_5align4typeE2ENS0_8appenderEcRZNS1_14do_write_floatIS5_NS1_9dragonbox10decimal_fpIdEEcNS1_14digit_groupingIcEEEET_SC_RKT0_RKNS0_12format_specsIT1_EENS1_11float_specsENS1_10locale_refEEUlS5_E3_EESD_SD_SK_yyOT2_: symbol wrong type (4 vs 3)
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot export _ZN3fmt3v106detail12write_paddedILNS0_5align4typeE2ENS0_8appenderEcRZNS1_14do_write_floatIS5_NS1_9dragonbox10decimal_fpIfEEcNS1_14digit_groupingIcEEEET_SC_RKT0_RKNS0_12format_specsIT1_EENS1_11float_specsENS1_10locale_refEEUlS5_E3_EESD_SD_SK_yyOT2_: symbol wrong type (4 vs 3)
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot export _ZSt9__find_ifIN9__gnu_cxx17__normal_iteratorIPN7precice4mesh6VertexESt6vectorIS4_SaIS4_EEEENS0_5__ops10_Iter_predIZNS2_7mapping4impl16createClusteringESt10shared_ptrINS3_4MeshEESG_djbEUlRT_E3_EEESH_SH_SH_T0_St26random_access_iterator_tag: symbol wrong type (4 vs 3)
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot export _ZZN3fmt3v106detail14do_write_floatINS0_8appenderENS1_14big_decimal_fpEcNS1_14digit_groupingIcEEEET_S7_RKT0_RKNS0_12format_specsIT1_EENS1_11float_specsENS1_10locale_refEENKUlS3_E3_clES3_: symbol wrong type (4 vs 3)
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot export _ZZN3fmt3v106detail14do_write_floatINS0_8appenderENS1_9dragonbox10decimal_fpIdEEcNS1_14digit_groupingIcEEEET_S9_RKT0_RKNS0_12format_specsIT1_EENS1_11float_specsENS1_10locale_refEENKUlS3_E3_clES3_: symbol wrong type (4 vs 3)
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot export _ZZN3fmt3v106detail14do_write_floatINS0_8appenderENS1_9dragonbox10decimal_fpIfEEcNS1_14digit_groupingIcEEEET_S9_RKT0_RKNS0_12format_specsIT1_EENS1_11float_specsENS1_10locale_refEENKUlS3_E3_clES3_: symbol wrong type (4 vs 3)
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot export _ZZNSt8__detail9_CompilerINSt7__cxx1112regex_traitsIcEEE18_M_expression_termILb0ELb0EEEbRNS4_13_BracketStateERNS_15_BracketMatcherIS3_XT_EXT0_EEEENKUlvE0_clEv: symbol wrong type (4 vs 3)
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot export _ZZNSt8__detail9_CompilerINSt7__cxx1112regex_traitsIcEEE18_M_expression_termILb0ELb1EEEbRNS4_13_BracketStateERNS_15_BracketMatcherIS3_XT_EXT0_EEEENKUlvE0_clEv: symbol wrong type (4 vs 3)
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot export _ZZNSt8__detail9_CompilerINSt7__cxx1112regex_traitsIcEEE18_M_expression_termILb1ELb0EEEbRNS4_13_BracketStateERNS_15_BracketMatcherIS3_XT_EXT0_EEEENKUlvE0_clEv: symbol wrong type (4 vs 3)
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot export _ZZNSt8__detail9_CompilerINSt7__cxx1112regex_traitsIcEEE18_M_expression_termILb1ELb1EEEbRNS4_13_BracketStateERNS_15_BracketMatcherIS3_XT_EXT0_EEEENKUlvE0_clEv: symbol wrong type (4 vs 3)
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.

```

@fsimonis I have good news and bad news 😔

I was able to build, and successfully link, the shared libraries and the tools using v3.2.0 with both Clang and GCC by applying the https://github.com/precice/precice/commit/f7a4723827bbf20062fe004322b837f348374af3 and https://github.com/precice/precice/commit/2d08cf23abe81490f11c2be0e00afd43648ee504 patches, but now building static libraries (`-DBUILD_SHARED_LIBS=OFF`) is not working. 

CMake fails with:
```
-- Found Python 3.12.10 with NumPy 2.2.5
-- Validating LibPython
-- Validating LibPython - success
-- Validating NumPy
-- Validating NumPy - success

-- === TARGETS & PACKAGES ===
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Excluding test sources
-- Compressing manpage: D:/Dev/Github/MINGW-packages/mingw-w64-precice/src/build-CLANG64-shared/packaging/man1/precice-tools.1
-- Debian package generator disabled: Install prefix is not "/usr"
-- Compressing changelog
-- Configuring done (8.2s)
CMake Error in CMakeLists.txt:
  install(EXPORT "preciceTargets" ...) includes target "precice" which
  requires target "preciceCore" that is not in any export set.


CMake Error in CMakeLists.txt:
  export called with target "precice" which requires target "preciceCore"
  that is not in any export set.


-- Generating done (0.1s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

--- a/.github/workflows/build-and-test-windows.yml
+++ b/.github/workflows/build-and-test-windows.yml
@@ -1,0 +1,79 @@
+name: Build and Test Windows MinGW
+on:
+  push:
+    branches:
+      - main
+      - develop
+  pull_request:
+    paths:
+      - 'cmake/**'
+      - 'examples/**'
+      - 'src/**'
+      - 'tests/**'
+      - 'thirdparty/**'
+      - 'tests/**'
+      - 'tools/profiling/**'
+      - 'CMakeLists.txt'
+      - 'extras/bindings/**'
+      - '.github/workflows/build-and-test-windows.yml'
+  workflow_dispatch:
+
+concurrency:
+  group: ${ {github.event_name }}-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{github.event_name == 'pull_request'}}
+
+jobs:
+  msys2-ucrt64:
+    name: ${{ format('MinGW UCRT {0}', matrix.TYPE) }}
+    runs-on: windows-latest
+    defaults:
+      run:
+        shell: msys2 {0}
+    strategy:
+      fail-fast: false
+      matrix:
+        TYPE: [Release, Debug]
+    env:
+      CMAKE_BUILD_PARALLEL_LEVEL: 4
+      CMAKE_BUILD_TYPE: ${{ matrix.TYPE }}
+      CMAKE_GENERATOR: Ninja
+      CTEST_OUTPUT_ON_FAILURE: True
+      CXXFLAGS: -Wa,-mbig-obj
+    steps:
+      - name: Install MS-MPI
+        run: |
+          $ProgressPreference = 'SilentlyContinue'
+          Invoke-WebRequest -Uri https://github.com/microsoft/Microsoft-MPI/releases/download/v10.1.1/msmpisetup.exe -OutFile msmpisetup.exe
+          Start-Process -FilePath .\msmpisetup.exe -ArgumentList '-unattend' -Wait
+          Remove-Item msmpisetup.exe
+
+          # Add MS-MPI to PATH
+          echo "C:\Program Files\Microsoft MPI\Bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+        shell: powershell
+      - uses: msys2/setup-msys2@v2
+        with:
+          msystem: UCRT64
+          update: true
+          path-type: inherit
+          install: git mingw-w64-ucrt-x86_64-gcc-libs mingw-w64-ucrt-x86_64-boost-libs mingw-w64-ucrt-x86_64-libxml2 mingw-w64-ucrt-x86_64-python mingw-w64-ucrt-x86_64-python-numpy mingw-w64-ucrt-x86_64-cc mingw-w64-ucrt-x86_64-cmake mingw-w64-ucrt-x86_64-msmpi mingw-w64-ucrt-x86_64-ninja mingw-w64-ucrt-x86_64-boost mingw-w64-ucrt-x86_64-eigen3 mingw-w64-ucrt-x86_64-crt mingw-w64-ucrt-x86_64-msmpi
+      - uses: actions/checkout@v3
+      - name: Make build directory
+        run: mkdir -p build
+      - name: Configure
+        working-directory: build
+        run: |
+          cmake -DPRECICE_FEATURE_PETSC_MAPPING=OFF -DPRECICE_FEATURE_PYTHON_ACTIONS=OFF -DCMAKE_CXX_FLAGS_DEBUG=-g1 ..
+      - name: Build
+        working-directory: build
+        run: |
+          cmake --build .
+      - name: Test
+        working-directory: build
+        run: |
+          # Allow each test to run twice to iron out sporadic MPI failures
+          ctest -j --repeat until-pass:2
+      - uses: actions/upload-artifact@v4
+        if: failure()
+        with:
+          name: debugdump
+          path: build/

